samples/drivers/led_sx1509b_intensity/src/main.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/gpio/gpio_sx1509b.h>
+
+#define GREEN_LED 5
+#define BLUE_LED 6
+#define RED_LED 7
+
+enum sx1509b_color { sx1509b_red = 0, sx1509b_green, sx1509b_blue };
+
+const struct device *sx1509b_dev;
+static struct k_delayed_work rgb_work;
+static uint8_t which_color = sx1509b_red;
+static uint8_t iterator;
+
+static void rgb_work_handler(struct k_work *work)
+{
+	if (which_color == sx1509b_red) {
+		sx1509b_led_intensity_pin_set(sx1509b_dev, RED_LED, iterator);
+
+		iterator++;
+
+		if (iterator >= 255) {
+			sx1509b_led_intensity_pin_set(sx1509b_dev, RED_LED, 0);
+			which_color = sx1509b_green;
+			iterator = 0;
+		}
+	} else if (which_color == sx1509b_green) {
+		sx1509b_led_intensity_pin_set(sx1509b_dev, GREEN_LED, iterator);
+
+		iterator++;
+
+		if (iterator >= 255) {
+			sx1509b_led_intensity_pin_set(sx1509b_dev, GREEN_LED,
+						      0);
+			which_color = sx1509b_blue;
+			iterator = 0;
+		}
+	} else if (which_color == sx1509b_blue) {
+		sx1509b_led_intensity_pin_set(sx1509b_dev, BLUE_LED, iterator);
+
+		iterator++;
+
+		if (iterator >= 255) {
+			sx1509b_led_intensity_pin_set(sx1509b_dev, BLUE_LED, 0);
+			which_color = sx1509b_red;
+			iterator = 0;
+		}
+	}
+
+	k_delayed_work_submit(&rgb_work, K_MSEC(10));
+}
+
+void main(void)
+{
+	int err;
+
+	printk("SX1509B intensity sample\n");
+
+	sx1509b_dev = device_get_binding(DT_PROP(DT_NODELABEL(sx1509b), label));
+
+	if (sx1509b_dev == NULL) {
+		printk("Error binding SX1509B device\n");
+
+		return;
+	}
+
+	for (int i = GREEN_LED; i <= RED_LED; i++) {
```

Comment:
You can't iterate over the pin numbers in general, because they may not be adjacent and in the right order.  Better to make this a static array above:
```
static const gpio_pin_t rgb_pins[] = {
   RED_LED,
   GREEN_LED,
   BLUE_LED,
};
```
ensuring that the indexes match the enumeration.  Then iterate over an integer that covers the color span and index for the pin.

I was confused that one ordering of leds (the pins) green/blue/red but the behavior is red/green/blue.  While fixing the above that inconsistency can be removed.